src/codeu/chat/client/simplegui/ChatSimpleGui.java

```diff
@@ -97,15 +94,14 @@ private void initialize() {
         labelPanel.add(new JLabel("Please choose from an option below:"));
         final GridBagConstraints labelViewC = new GridBagConstraints();
 
-        final JPanel userSignIn = new SignInPanel();
+        // Creates new Jpanel for user sign in pane
+        final JPanel userSignIn = new SignInPanel(clientContext);
         userSignIn.setBorder(paneBorder());
-        codeu.chat.client.simplegui.SignInPanel.setFrame(mainFrame);
         final GridBagConstraints usersViewC = new GridBagConstraints();
 
         // ConversationsPanel gets access to MessagesPanel
-        final CreateAccountPanel newAccountPanel = new CreateAccountPanel();
+        final CreateAccountPanel newAccountPanel = new CreateAccountPanel(clientContext);
```

Comment:
why is everything final....? local variables don't need to be final unless you're using them in an anonymous class (i.e. an actionhandler or something)